### PR TITLE
Fix compilation issues while targeting Q.

### PR DESCRIPTION
**Pull request by MatthewTighe**
## What changes does this PR introduce?
This PR fixes compilation issues that occurred while targeting Q. They fell into two main categories:
- Navigation
- Increased number of correctly annotated nullable JVM types

I also took the time to do some general cleanup while I was rooting around.

## Any background context you want to provide?
App would not compile targeting SDK 29.

## Where should the reviewer start?
Probably with the navigation stuff in `nav_graph.xml` and the various fragments.

## Has this been manually tested? How?
Yes, by navigating around and starting a session.

## What value does this provide to our end users?
Allows us to target Q and remain up-to-date.

### CODE REVIEW

File: app/src/main/java/tech/ula/utils/QWarningHandler.kt [hunk 1]
```diff
@@ -28,7 +28,8 @@ class QWarningHandler(private val prefs: SharedPreferences, private val ulaFiles
     }
 
     private fun userHasFilesystems(): Boolean {
-        val hasFilesystems = ulaFiles.filesDir.listFiles().mapNotNull { it.name.toIntOrNull() }.isNotEmpty()
+        val files = ulaFiles.filesDir.listFiles() ?: return false
```
Comment (luongthomas) [inline app/src/main/java/tech/ula/utils/QWarningHandler.kt]: That's neat that you can return in a ternary operation

File: app/src/test/java/tech/ula/utils/ServerUtilityTest.kt [hunk 2]
```diff
@@ -23,15 +23,15 @@ class ServerUtilityTest {
```
Comment (luongthomas) [inline app/src/test/java/tech/ula/utils/ServerUtilityTest.kt]: This should be removed since the `LogUtility` class was removed

Comment (luongthomas) [reply app/src/test/java/tech/ula/utils/ServerUtilityTest.kt]: All references to LogUtility in this test file should be removed, that may be the cause of CI failing.

Comment (MatthewTighe) [reply app/src/test/java/tech/ula/utils/ServerUtilityTest.kt]: Yep, that should fix it.

File: app/src/main/java/tech/ula/utils/Logger.kt [hunk 3]
```diff
@@ -85,12 +92,14 @@ class SentryLogger : Logger {
...
         Sentry.capture(event)
+        Log.e("ILLEGAL_STATE", message)
     }
 
     override fun sendEvent(message: String) {
         val event = EventBuilder()
                 .withMessage(message)
                 .withLevel(Event.Level.ERROR)
         Sentry.capture(event)
+        Log.e("EVENT", message)
```
Comment (luongthomas) [inline app/src/main/java/tech/ula/utils/Logger.kt]: Looks like this is causing a failing test, complaining that `Log.e` hasn't been mocked in `FilesystemListViewModelTest`

**Pull request merged**